pandora.py

```diff
@@ -206,6 +207,25 @@ def __init__(self):
         })
         return Station(response)
 
+    async def update_station(self, station: Station, name: Union[str, None]=None,
+                             description: Union[str, None]=None) -> None:
+        """
+        Update a station's name and/or description.
+
+        :param station: The station to be renamed.
+        :param name: The new name of the station or None to not change it.
+        :param description: The new description of the station or None to not change it.
+        """
+        if name and len(name) > 64:
+            name = name[:63] + '…'
+        if description and len(description) > 4000:
+            description = description[:3999] + '…'
+        await self._send_message('v1/station/updateStation', {
+            'stationId': station.station_id,
+            'name': name or station.name,
```

Comment:
I would make the dict ahead of time and only insert the values if needed:

```python
body = {}
if name is not None:
    body['name'] = name
if description is not None:
   body['description'] = description
 await self._send_message('v1/station/updateStation', body)
```

Also I would factor out your ellipsizing into a function so you could just do:

```python
    body['name'] = ellipsize(name, 64)
```

Reply:
I can factor out the ellipsize but in the web client both fields are always sent. It will work with just one but then you can end up with a situation where it's possible to try to set the name to an empty string (' ' != None)